lam4doc/services/handlers.py

```diff
@@ -5,52 +5,99 @@
 
-    logger.debug('finish service for preparing lam template ')
+    logger.debug(f'finish service for generating report with target_location: {target_location}')
+    return report_path
 
 
-def generate_lam_report(target_location: str, report_builder: ReportBuilder) -> Path:
+def generate_lam_report(target_location: str) -> Path:
     """
-    Handler for generating the lam report.
+    Method for generating the lam report using the generic prepare_template and generate_report
     :param target_location: path to the report templates
-    :param report_builder: a report builder type service (eds4jinja2.builders.report_builder.ReportBuilder)
-    :return: location of the report document
+    :return: path to report
     """
-    logger.debug('start service for generating lam report ')
+    logger.debug('start service for generating lam report')
 
-    report_builder.make_document()
+    report_builder = ReportBuilder(target_path=config.LAM_REPORT_TEMPLATE_LOCATION,
+                                   output_path=target_location,
+                                   additional_config={"conf": {"default_endpoint": config.LAM_FUSEKI_SERVICE}})
+    report_location = generate_report(target_location, report_builder)
 
-    report_path = Path(str(target_location)) / 'output/main.html'
+    logger.debug('finish service for generating lam report')
+    return report_location
 
-    logger.debug('finish service for generating lam report ')
-    return report_path
+
+def generate_indexes(target_location: str) -> List[FileInfo]:
+    """
+    Method for generating the lam indexes and zipping them using the generic prepare_template and generate_report
+    :param target_location: path to where to create the reports
+    :return: list of indexes as FileInfo tuples
+    """
+    logger.debug('start service for generating lam indexes')
+
+    index_info = [IndexInfo('celex', config.LAM_CELEX_CONFIG_NAME),
+                  IndexInfo('classes', config.LAM_CLASSES_CONFIG_NAME),
+                  IndexInfo('properties', config.LAM_PROPERTIES_CONFIG_NAME)]
+    index_files_info = list()
+
+    for index in index_info:
+        logger.debug(f'create report for {index.name}, with config {index.config_location}')
+        index_target_location = Path(target_location) / index.name
+        index_target_location.mkdir()
+        report_builder = ReportBuilder(target_path=config.LAM_INDEXES_TEMPLATE_LOCATION,
+                                       config_file=index.config_location,
+                                       output_path=index_target_location,
+                                       additional_config={"conf": {"endpoint": config.LAM_FUSEKI_SERVICE}})
+        index_files_info.append(FileInfo(location=generate_report(index_target_location, report_builder),
+                                         file_name=f'{index.name}.json'))
+
+    logger.debug('finish service for generating lam indexes')
```

Comment:
```suggestion
    logger.debug('finished generating indexes')
```

Reply:
everywhere else (in the other projects) I used the present simple tense verbs. I can change them if you wish. 
I used `service` to differentiate what action is happening. If I remove the `service` from the logs, it will be more difficult to understand, from the logs, where it came from